### PR TITLE
Aimmo unity communication

### PR DESCRIPTION
**Pull request by mrniket**
I also removed the map feature 2D drawing code and cleaned up the code base a little. This PR should be merged in with aimmo #363

### CODE REVIEW

File: UnityProject/Assets/Scripts/Players/PlayerController.cs [hunk 1]
```diff
@@ -1,70 +1,73 @@
...
+        public void SetNextState(PlayerDTO playerDTO)
+        {
+            nextState = playerDTO;
+
+            transform.GetComponent<IsometricPosition>().Set(nextPosition);
+            nextPosition = new Vector2(nextState.location.x, nextState.location.y);
+        }
+
+    }
+}
```
Comment (CelineBoudier) [inline UnityProject/Assets/Scripts/Players/PlayerController.cs]: missing new line?

File: UnityProject/Assets/Scripts/Players/PlayerHealthBar.cs [hunk 2]
```diff
@@ -1,158 +1,161 @@
...
-	// Method that calculates how much we need to move the green bar when its
-	// scale changes.
-	private float GetGreenBarX(float hp)
-	{
-		return GreenBarPositionZeroHP + (hp * GreenBarIncrementOneHP);
-	}
+
+    /* Creates the health bar sprite attached to each avatar and provides an easy
+     * interface to modify it. When the health points change, it suffices to call
+     * SetHealthPoints() and this class will do the rest.ss
```
Comment (CelineBoudier) [inline UnityProject/Assets/Scripts/Players/PlayerHealthBar.cs]: ??

File: UnityProject/Assets/Scripts/WorldControls.cs [hunk 3]
```diff
@@ -148,71 +131,30 @@ public void EstablishConnection()
...
-
-		foreach (GameObject mapFeature in allMapFeatures) 
-			Destroy(mapFeature);
-		
-		foreach (GameObject avatar in allAvatars) 
-			Destroy(avatar);
-		
+        playerManager.OverwritePlayersState(players);
 	}
 }
```
Comment (CelineBoudier) [inline UnityProject/Assets/Scripts/WorldControls.cs]: extra new line?

Comment (mrniket) [reply UnityProject/Assets/Scripts/WorldControls.cs]: where is the extra new line here?

Comment (mrniket) [reply UnityProject/Assets/Scripts/WorldControls.cs]: done :)

File: UnityProject/Assets/Scripts/Players/PlayerController.cs (hunk 1 above)
Comment (mrniket) [reply UnityProject/Assets/Scripts/Players/PlayerController.cs]: done

File: UnityProject/Assets/Editor/Tests/DTODeserialisationTests.cs [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+            Assert.AreEqual(Era.LessFlat, gameState.EraType);
+            Assert.AreEqual(new Location(-2, -2), gameState.southWestCorner);
+            Assert.AreEqual(new Location(2, 2), gameState.northEastCorner);
+            Assert.AreEqual(1, gameState.players.Length);
+            Assert.AreEqual(1, gameState.pickups.Length);
+            Assert.AreEqual(1, gameState.scoreLocations.Length);
+            Assert.AreEqual(1, gameState.obstacles.Length);
+        }
+    }
+}
```
Comment (OlafSzmidt) [inline UnityProject/Assets/Editor/Tests/DTODeserialisationTests.cs]: New line

File: UnityProject/Assets/Scripts/Players/PlayerHealthBar.cs [hunk 5]
```diff
@@ -1,158 +1,161 @@
...
+        private const float MaxHP = 10.0f;
+        private const float MinHPFromBackend = 0.0f;
+        private const float MaxHPFromBackend = 10.0f;
+        private const float GreenBarPositionZeroHP = -0.5f;
+        private const float GreenBarIncrementOneHP = 0.05f;
+
+        // Health bar game objects.
+        private GameObject healthBar;
+        private GameObject redBar;
+        private GameObject greenBar;
```
Comment (OlafSzmidt) [inline UnityProject/Assets/Scripts/Players/PlayerHealthBar.cs]: Matter of preference really, but with this amount of constants we might as well create a separate package for them, and a class for each type?

File: UnityProject/Assets/Scripts/Players/PlayerScoreText.cs [hunk 6]
```diff
@@ -1,45 +1,46 @@
...
+        // Text positioning.
+        private const float ScaleX = 0.1f;
+        private const float ScaleY = 0.1f;
+        private const float ScaleZ = 0.1f;
+
+        private const float OffsetX = 0.0f;
+        private const float OffsetY = 1.5f;
+        private const float OffsetZ = 0.0f;
+
+        private const int fontSize = 80;
```
Comment (OlafSzmidt) [inline UnityProject/Assets/Scripts/Players/PlayerScoreText.cs]: Maybe in a new class?

File: UnityProject/Assets/Editor/Tests/DTODeserialisationTests.cs [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+            }";
+
+            GameStateDTO gameState = JsonUtility.FromJson<GameStateDTO>(gameUpdateJSON);
+            Assert.AreEqual(Era.LessFlat, gameState.EraType);
+            Assert.AreEqual(new Location(-2, -2), gameState.southWestCorner);
+            Assert.AreEqual(new Location(2, 2), gameState.northEastCorner);
+            Assert.AreEqual(1, gameState.players.Length);
+            Assert.AreEqual(1, gameState.pickups.Length);
+            Assert.AreEqual(1, gameState.scoreLocations.Length);
+            Assert.AreEqual(1, gameState.obstacles.Length);
```
Comment (OlafSzmidt) [inline UnityProject/Assets/Editor/Tests/DTODeserialisationTests.cs]: `Length` is with upper case, `EraType` is with upper case, `northEastCorner` isn't. What's the convention here?

File: UnityProject/Assets/Scripts/Players/PlayerController.cs [hunk 8]
```diff
@@ -1,70 +1,73 @@
...
+        private const float moveInterval = 0.5f;
 
-	// Initialisation.
-	public void Awake()
-	{
-		startTime = Time.time;
-		currPosition = transform.position;
-		nextState = new PlayerData(transform.position);
-	}
+        private float startTime;
```
Comment (OlafSzmidt) [inline UnityProject/Assets/Scripts/Players/PlayerController.cs]: Consider creating a separate package with classes of each category for constants. We seem to use a looooooooooooooot.

File: UnityProject/Assets/Editor/Tests/PlayerControllerTest.cs [hunk 9]
```diff
@@ -31,8 +29,9 @@ public Vector2 GetPosition()
 			}
 
 			public void SetNextPosition(float x, float y)
-			{	
-				PlayerData playerData = new PlayerData(new Vector2(x, y));
+			{
+                PlayerDTO playerData = new PlayerDTO();
+                playerData.location = new Location((int)x, (int)y);
```
Comment (OlafSzmidt) [inline UnityProject/Assets/Editor/Tests/PlayerControllerTest.cs]: Fix indentation

**Pull request merged**